hs628_tools/regression_check.py

```diff
@@ -0,0 +1,189 @@
+# coding: utf-8
+
+import numpy as np
+import pandas as pd
+import sklearn as sk
+from sklearn.linear_model import LinearRegression
+import statsmodels.api as sm
+import matplotlib.pyplot as plt
+import seaborn as sns
+import os
+import sys
+# %config IPCompleter.greedy=True
+# %matplotlib inline
+
+#   ****** Data Visualization to check Linear regression assumptions and calculating new regression metric ******
+
+class RegCheck():
+    """
+        This Class is created for checking Linear Regression assumptions and Evaluating the Model
+        The following functions are defined in this class :
+
+         func : check_linearity_residual_homoscedasticity_independence_plt checks to see if the
+                data can fit to a linear model/non-linear model. Also check the homoscedasticity
+                (check for the spread of data) of the residuals/error terms and also check for their
+                independence. We diagnose these through a Residuals vs Predicted values plot.If there
+                is no clear pattern and data symmetrically distributed it can be said to satisfy the
+                assumptions.
+
+         func : check_residuals_normality_plt checks to see the normality of the error terms through
+                a density plot of the residuals. If the graph looks like gaussian normal curve it can
+                be said that the residuals satisfy the assumption of normality.
+
+         func : check_predictors_independence_plt checks for the independence of each predictor by
+                plotting a heatmap of the correlations of each predictor with one and another. The
+                heatmap would clearly indicate if there are multicolinear predictors.
+
+         func : regression_metric builds a metric for evaluating the model. It is a harmonic mean of
+                the R-Squared value and the F-Statistic taking into consideration the probability of
+                significance of the F-statistic being significant than having an only intercept model.
+    """
+
+    def __init__(self):
+        # Instantiating the parameters used in the methods below :PredictorColumnnames,R2,
+        # F-stat,F-statpval,ResidualSumofSquares,RegressionMetric
+        self.colnames = None
+        self.rsquared = 0
+        self.fstatistic = 0
+        self.fstat_pval = 0
+        self.resid_sumsq = 0
+        self.regmetric = 0
+
+    def check_linearity_residual_homoscedasticity_independence_plt(self, X, y, model):
+        """
+        Checks to see if the data can fit to a linear model/non-linear model,checks the homoscedasticity
```

Comment:
It looks to me like this method does not actually check these things itself, but instead displays a plot that help the user check such things, so the docstring should reflect that (and ideally either explain how to interpret the plots or include a link to someplace that does).

Reply:
Will add the changes in docstrings to reflect its just a plot from which user needs to infer. About how to infer the plots, I have already added that above in the docstrings for the class and the functions in the class.